src/components/QueryEditor/LuceneQueryEditor/state.ts

```diff
@@ -0,0 +1,21 @@
+import { type Action, createAction } from '@reduxjs/toolkit';
+import { initQuery } from '../state';
+import { LuceneQueryType, OpenSearchQuery } from 'types';
+
+export const UPDATE_LUCENE_TYPE_AND_METRICS = 'update_lucene_type_and_metrics';
+
+export const updateLuceneTypeAndMetrics = createAction<{
```

Comment:
`updateLuceneTypeAndMetrics` updates two reducers (this one and `metricsReducer`) at the same time,. 
 This is because, if I triggered `onChange(query)` from the component and then `changeMetricType`, one would overwrite the other (`changeMetricType` calls `onChange` in the background)